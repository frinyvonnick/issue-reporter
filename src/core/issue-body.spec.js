const makeIssueBodyFromReport = require('./issue-body')

describe('makeIssueBodyFromReport', () => {
  it('should format a json error as markdown', () => {
    const errorReport = {
      error: {
        title: 'Stacktrace',
        stack: 'Stack'
      },
      environment: {
        title: 'Environment',
        markdown: 'Environment markdown'
      }
    }

    const result = makeIssueBodyFromReport({ errorReport })

    expect(result).toMatchSnapshot()
  })

  it('should format a json error as markdown', () => {
    const errorReport = {
      error: {
        title: 'Stacktrace',
        stack: 'Stack'
      },
      environment: {
        title: 'Environment',
        markdown: 'Environment markdown'
      }
    }

    const result = makeIssueBodyFromReport({
      errorReport,
      sections: [
        {
          title: 'Custom section',
          content: {
            version: '1.0.1',
            release: '2.0.0',
            commit: 'qwej79qwjeqw8euqo8wj8eq8'
          }
        },
        {
          title: 'Second section',
          content: {
            'awesome info': 'it works'
          }
        }
      ]
    })

    expect(result).toMatchSnapshot()
  })

  it('should call user format function', () => {
    const errorReport = {
      error: {
        title: 'Stacktrace',
        stack: 'Stack'
      },
      environment: {
        title: 'Environment',
        markdown: 'Environment markdown'
      }
    }

    const customFormat = 'Custom format'
    const formatReport = jest.fn(() => customFormat)

    const result = makeIssueBodyFromReport({ errorReport, formatReport })

    expect(formatReport).toHaveBeenCalledWith(errorReport)
    expect(result).toBe(customFormat)
  })

  it('should handle empty sections', () => {
    const errorReport = {
      error: {
        title: 'Stacktrace',
        stack: 'Stack'
      },
      environment: {
        title: 'Environment',
        markdown: 'Environment markdown'
      }
    }

    const result = makeIssueBodyFromReport({
      errorReport,
      sections: [
        {
          title: 'Custom section',
          content: null
        },
        {
          title: 'Second section',
          content: {
            'awesome info': 'it works'
          }
        }
      ]
    })

    expect(result).toMatchSnapshot()
  })
})
