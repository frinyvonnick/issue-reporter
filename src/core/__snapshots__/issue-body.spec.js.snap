// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`makeIssueBodyFromReport should format a json error as markdown 1`] = `
"## Stacktrace

\`\`\`
Stack
\`\`\`

## Environment

Environment markdown
"
`;

exports[`makeIssueBodyFromReport should format a json error as markdown 2`] = `
"## Stacktrace

\`\`\`
Stack
\`\`\`

## Environment

Environment markdown

## Custom section

- **version**: 1.0.1
- **release**: 2.0.0
- **commit**: qwej79qwjeqw8euqo8wj8eq8

## Second section

- **awesome info**: it works
"
`;

exports[`makeIssueBodyFromReport should handle empty sections 1`] = `
"## Stacktrace

\`\`\`
Stack
\`\`\`

## Environment

Environment markdown

## Second section

- **awesome info**: it works
"
`;
